src/test/java/logic/MainLogicImplTest.java

```diff
@@ -0,0 +1,80 @@
+package logic;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+public class MainLogicImplTest {
+
+    private static final String TWO_FRUITS_FILE
+            = "src/test/resources/IncomingReports/TwoFruits.csv";
+    private static final String THREE_FRUITS_FILE
+            = "src/test/resources/IncomingReports/ThreeFruits.csv";
+    private static final String REPORT_FILE
+            = "src/test/resources/OutGoingReports/NormReport.csv";
+    private static final String BUILT_IN_REPORT_PATH
+            = "src/main/resources/OutGoingReports/Report.csv";
+
+    @Test
+    public void endToEnd_twoFruits_ok() {
+        List<String> expectedReportLines = new ArrayList<>();
+        expectedReportLines.add("fruit,quantity");
+        expectedReportLines.add("banana,152");
+        expectedReportLines.add("apple,90");
+        new MainLogicImpl(TWO_FRUITS_FILE, REPORT_FILE).generateReport();
+        List<String> actualReportLines;
+        try {
+            actualReportLines = Files.readAllLines(Path.of(REPORT_FILE));
+        } catch (IOException e) {
+            throw new RuntimeException("Test failed with internal error, can't read file "
+                    + REPORT_FILE + e);
+        }
```

Comment:
Why do you have a try catch inside the test?

Reply:
We should avoid usage of existing methods in our program while testing other methods